pkg/reconciler/v1alpha1/pipelinerun/resources/conditionresolution.go

```diff
@@ -95,24 +99,65 @@ func (rcc *ResolvedConditionCheck) ConditionToTaskSpec() *v1alpha1.TaskSpec {
 		Steps: []corev1.Container{rcc.Condition.Spec.Check},
 	}
 
-	if len(rcc.Condition.Spec.Params) > 0 {
-		t.Inputs = &v1alpha1.Inputs{
-			Params: rcc.Condition.Spec.Params,
-		}
-		// convert param strings of type ${params.x} to ${inputs.params.x}
-		// in order to apply taskrun substitution
-		convertParamTemplates(&t.Steps[0], rcc.Condition.Spec.Params)
+	t.Inputs = &v1alpha1.Inputs{
+		Params:    rcc.Condition.Spec.Params,
+		Resources: toTaskResouces(rcc.Condition.Spec.Resources),
+	}
+
+	// convert param strings of type ${params.x} to ${inputs.params.x}
+	convertParamTemplateStrings(&t.Steps[0], rcc.Condition.Spec.Params, rcc.Condition.Spec.Resources)
+	// convert resource strings of type ${resources.name.key} to ${inputs.resources.name.key}
+	err := convertResourceTemplateStrings(&t.Steps[0], rcc.ResolvedResources, rcc.Condition.Spec.Resources)
+
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to replace resource template strings %w", err)
+	}
+
+	return t, nil
+}
+
+func toTaskResouces(resources []v1alpha1.ConditionResource) []v1alpha1.TaskResource {
```

Comment:
might not need this if we can use the same type (ish?? see my comment above) for both this and TaskResources :D

Reply:
if we _do_ still need this function - maybe it would make sense to have this as a function on the `ConditionResource` itself, in the `_types.go` file?